chapters/data-analysis.tex

```diff
@@ -1,292 +1,488 @@
+Automating them will save you time by the end of the process. 
+However, don't spend too much time formatting tables and graphs until you are ready to publish.\sidenote{For a more detailed discussion on this, including different ways to export tables from Stata, see \url{https://github.com/bbdaniels/stata-tables}}
+Polishing final outputs can be a time-consuming process, 
+and you want to it as few times as possible.
+
+We cannot stress this enough: don't ever set a workflow that requires copying and pasting results from the console.
+There are numerous commands to export outputs from both R and Stata.\sidenote{Some examples are \href{ http://repec.sowi.unibe.ch/stata/estout/}{\texttt{estout}}, \href{https://www.princeton.edu/~otorres/Outreg2.pdf}{\texttt{outreg2}}, 
+and \href{https://www.benjaminbdaniels.com/stata-code/outwrite/}{\texttt{outwrite}} in Stata, 
+and \href{https://cran.r-project.org/web/packages/stargazer/vignettes/stargazer.pdf}{\texttt{stargazer}}
+and \href{https://ggplot2.tidyverse.org/reference/ggsave.html}{\texttt{ggsave}} in R.}
+Save outputs in accessible and, whenever possible, lightweight formats.
+Accessible means that it's easy for other people to open them.
+In Stata, that would mean always using \texttt{graph export} to save images as \texttt{.jpg}, \texttt{.png}, \texttt{.pdf}, etc., 
+instead of \texttt{graph save}, which creates a \texttt{.gph} file that can only be opened through a Stata installation.
+Some publications require ``lossless'' TIFF of EPS files, which are created by specifying the desired extension.
+For tables, \texttt{.tex} is preferred. 
+Excel \texttt{.xlsx} and \texttt{.csv} files are also acceptable, 
+although if you are working on a large report they will become cumbersome to update after revisions.
+Whichever format you decide to use, remember to always specify the file extension explicitly.
 
+% Formatting
+If you need to create a table with a very particular format, that is not automated by any command you know, consider writing the it manually 
+(Stata's \texttt{filewrite}, for example, allows you to do that).
+This will allow you to write a cleaner script that focuses on the econometrics, and not on complicated commands to create and append intermediate matrices.
+To avoid cluttering your scripts with formatting and ensure that formatting is consistent across outputs,
+define formatting options in an R object or a Stata global and call them when needed.
+% Output content
+Keep in mind that final outputs should be self-standing.
+This means it should be easy to read and understand them with only the information they contain.
+Make sure labels and notes cover all relevant information, such as sample, unit of observation, unit of measurement and variable definition.\sidenote{\url{https://dimewiki.worldbank.org/wiki/Checklist:\_Reviewing\_Graphs} \\ \url{https://dimewiki.worldbank.org/wiki/Checklist:\_Submit\_Table}}
 
```

Comment:
Let's have a closing sentence or two that links to the next chapter, something along the lines of once data analysis is complete you're ready to publish (but hopefully better than that!)

Reply:
First attempt in 529aa73